cli/cmd/deploy/deploy_test.go

```diff
@@ -0,0 +1,204 @@
+
+	// create a rill project
+	tempDir := t.TempDir()
+	err := os.WriteFile(filepath.Join(tempDir, "rill.yaml"), []byte(`compiler: rillv1
+display_name: Untitled Rill Project
+olap_connector: duckdb`), 0644)
+	require.NoError(t, err)
+
+	// create a github repo
+	repo, _, err := ghClient.Repositories.Create(t.Context(), "", &github.Repository{
+		Name:    github.Ptr("self-hosted-git-repo" + uuid.NewString()[:8]),
+		Private: github.Ptr(true),
+	})
+	require.NoError(t, err)
+
+	// cleanup repo
+	t.Cleanup(func() {
+		owner, ghrepo, ok := gitutil.SplitGithubRemote(*repo.CloneURL)
+		require.True(t, ok, "invalid github remote: %s", *repo.CloneURL)
+		_, err = ghClient.Repositories.Delete(context.Background(), owner, ghrepo)
+		require.NoError(t, err, "failed to delete github repo %s/%s: %w", owner, ghrepo, err)
+	})
+
+	// push to github
+	require.NoError(t, err)
+	author := &object.Signature{
+		Name:  "Rill test user",
+		Email: "test.user@rilldata.com",
+	}
+	require.NoError(t, err)
+	err = gitutil.CommitAndForcePush(t.Context(), tempDir, &gitutil.Config{
+		Remote:        *repo.CloneURL,
+		DefaultBranch: "main",
+	}, "", author)
+	require.NoError(t, err, "failed to push to github repo")
+
+	// deploy project backed by github
+	result = adm.Run(t, "deploy", "--interactive=false", "--org=github-test", "--project=self-hosted-deploy", "--skip-deploy=true", "--path="+tempDir)
+	require.Equal(t, 0, result.ExitCode, result.Output)
+
+	// verify the project is correctly created
+	resp, err := adminClient.GetProject(t.Context(), &adminv1.GetProjectRequest{
+		OrganizationName: "github-test",
+		Name:             "self-hosted-deploy",
+	})
+	require.NoError(t, err)
+	require.Equal(t, "self-hosted-deploy", resp.Project.Name)
+	require.True(t, resp.Project.ManagedGitId == "")
+
+	// check remote configured in directory
+	remote, err := gitutil.ExtractGitRemote(tempDir, "origin", false)
+	require.NoError(t, err)
+	require.Equal(t, *repo.CloneURL, remote.URL)
+
+	// redeploy the same project with changes
+	err = os.Mkdir(filepath.Join(tempDir, "models"), 0755)
+	require.NoError(t, err)
+	err = os.WriteFile(filepath.Join(tempDir, "models/model.sql"), []byte(`SELECT 1 AS one`), 0644)
+	require.NoError(t, err)
+	result = adm.Run(t, "deploy", "--interactive=false", "--org=github-test", "--project=self-hosted-deploy", "--skip-deploy=true", "--path="+tempDir)
+	require.Equal(t, 0, result.ExitCode, result.Output)
+
+	// verify changes are pushed to Github repo
+	verifyGithubRepoContents(t, ghClient, resp.Project.GitRemote)
+}
+
+func verifyGithubRepoContents(t *testing.T, client *github.Client, remote string) {
+	owner, repo, ok := gitutil.SplitGithubRemote(remote)
+	require.True(t, ok, "invalid github remote: %s", remote)
+
+	con, _, _, err := client.Repositories.GetContents(t.Context(), owner, repo, "models/model.sql", nil)
+	require.NoError(t, err)
+	contents, err := con.GetContent()
+	require.NoError(t, err)
+	require.Equal(t, "SELECT 1 AS one", contents)
```

Comment:
Instead of hard-coding one file here, if you have a map of files as per my comment above, you could pass that map in here and iterate over it, validating that each file is in the repo.